common/gcloud.py

```diff
@@ -130,10 +130,9 @@ def run_local_instance(startup_script: Optional[str] = None) -> bool:
     """Does the equivalent of "create_instance" for local experiments, runs
     |startup_script| in the background."""
     command = ['/bin/bash', startup_script]
-    with subprocess.Popen(command,
-                          stdout=subprocess.PIPE,
-                          stderr=subprocess.STDOUT):
-        return new_process.ProcessResult(0, '', False)
+    # pylint: disable=consider-using-with
+    subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    return new_process.ProcessResult(0, '', False).retcode == 0
```

Comment:
This will always be true wont it?

Reply:
This reverts to https://github.com/google/fuzzbench/pull/1529/files#diff-f61c598f4891541e693644bbc9016e5841fcddc986ea967eb3e192586a8c57deL134

Reply:
Hmm, maybe I can also fix it by adding `==0` to `return new_process.ProcessResult(0, '', False)`.

Reply:
I don't think the new code (after revert) will always be true.
For example, it will be false if the process fails and returns non-zero value, right?

Reply:
The way it was before using a context manager was:
```
subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
return new_process.ProcessResult(0, '', False)
```
That indeed does not check anything about Popen.  `new_process.ProcessResult(0, '', False)` is just creating a named tuple (see definition here https://github.com/google/fuzzbench/blob/master/common/new_process.py#L67). There is no connection between that line and the call to `Popen`.

The important fix here though is to stop using the context manager. As per the documentation (https://docs.python.org/3/library/subprocess.html?highlight=popen#subprocess.Popen)

>Popen objects are supported as context managers via the [with](https://docs.python.org/3/reference/compound_stmts.html#with) statement: on exit, standard file descriptors are closed, and the process is waited for.

Using a context manager closes file descriptors and waits for the process to terminate, which is probably what is causing problems. This code probably needs to start the process, but not wait for its termination.

My suggestion would be:
```
subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
return True
```
If we want to be extra careful, we should check that the subprocess is created successfully, but I think we will get an exception if that were not the case anyway.

Reply:
Thanks, @aeflores!
Yeah, your suggestion is cleaner as the old return statement is trivially true.
I've fixed it accordingly : )